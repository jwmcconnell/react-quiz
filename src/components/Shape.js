import React from 'react';
import PropTypes from 'prop-types';

const Shape = ({ text, color }) => {
  return (
    <div style={{ width: '100px', height: '100px', color: color }}>
      <span>{text}</span>
    </div>
  );
};

Shape.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired
};

export default Shape;
