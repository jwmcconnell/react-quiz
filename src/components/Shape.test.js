import React from 'react';
import { shallow } from 'enzyme';
import Shape from './Shape';

describe('Shape component', () => {
  it('renders Shape', () => {
    const wrapper = shallow(<Shape text='' color='#FFFFFF' backgroundColor='#000000' />);
    expect(wrapper).toMatchSnapshot();
  });
});
