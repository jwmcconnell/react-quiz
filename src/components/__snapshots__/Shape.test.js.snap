// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape component renders Shape 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#000000",
      "color": "#FFFFFF",
      "height": "100px",
      "width": "100px",
    }
  }
>
  <span />
</div>
`;
